tools/pd-ctl/pdctl/ctl_command_test.go

```diff
@@ -0,0 +1,399 @@
+	suite.cluster = mustNewCluster(ctx, re, 1)
+	suite.cancel = cancel
+}
+
+func (suite *commandTestSuite) TearDownSuite() {
+	suite.cancel()
+	for _, server := range suite.cluster {
+		server.Close()
+		testutil.CleanServer(server.GetConfig().DataDir)
+	}
+}
+func (suite *commandTestSuite) TestClusterCommand() {
+	var (
+		re     = suite.Require()
+		bufOut = &bytes.Buffer{}
+		bufErr = &bytes.Buffer{}
+		addr   = suite.cluster[0].GetAddr()
+	)
+
+	// $ pdctl cluster --pd <addr>
+	args := fmt.Sprintf("cluster --pd %s", addr)
+	cmd := GetRootCmd()
+	cmd.SetArgs(strings.Split(args, " "))
+	cmd.SetOut(bufOut)
+	cmd.SetErr(bufErr)
+
+	err := cmd.Execute()
+	re.NoError(err)
+	re.Empty(bufErr)
+
+	var actual metapb.Cluster
+	err = json.Unmarshal(bufOut.Bytes(), &actual)
+	re.NoError(err)
+	re.Equal(suite.cluster[0].ClusterID(), actual.GetId())
+
+	// $ pdctl cluster status --pd <addr>
+	args = fmt.Sprintf("cluster status --pd %s", addr)
+	cmd.SetArgs(strings.Split(args, " "))
+
+	bufOut.Reset()
+	bufErr.Reset()
+
+	err = cmd.Execute()
+	re.NoError(err)
+	re.Empty(bufErr)
+
+	var actualStatus cluster.Status
+	err = json.Unmarshal(bufOut.Bytes(), &actualStatus)
+	re.NoError(err)
+	suite.True(actualStatus.IsInitialized)
+}
+
+func (suite *commandTestSuite) TestConfigCommand() {
+	var (
+		re     = suite.Require()
+		bufOut = &bytes.Buffer{}
+		bufErr = &bytes.Buffer{}
+		addr   = suite.cluster[0].GetAddr()
+	)
+
+	// $ pdctl show --pd <addr>
+	args := fmt.Sprintf("config show all --pd %s", addr)
+	cmd := GetRootCmd()
+	cmd.SetArgs(strings.Split(args, " "))
+	cmd.SetOut(bufOut)
+	cmd.SetErr(bufErr)
+
+	err := cmd.Execute()
+	re.NoError(err)
+	re.Empty(bufErr)
+
+	var actual config.Config
+	err = json.Unmarshal(bufOut.Bytes(), &actual)
+	re.NoError(err)
+
+	expected := suite.cluster[0].GetConfig()
+	// this assignment is required per https://github.com/tikv/pd/blob/6c76647809cc73368820d17470554a9943ab9422/server/api/config.go#L58
+	expected.Schedule.MaxMergeRegionKeys = expected.Schedule.GetMaxMergeRegionKeys()
+	re.Equal(expected.String(), strings.TrimSpace(bufOut.String()))
+}
+
+func (suite *commandTestSuite) TestHealthCommand() {
+	var (
+		re     = suite.Require()
+		bufOut = &bytes.Buffer{}
+		bufErr = &bytes.Buffer{}
+		addr   = suite.cluster[0].GetAddr()
+	)
+
+	// $ pdctl health --pd <addr>
+	args := fmt.Sprintf("health --pd %s", addr)
+	cmd := GetRootCmd()
+	cmd.SetArgs(strings.Split(args, " "))
+	cmd.SetOut(bufOut)
+	cmd.SetErr(bufErr)
+
+	err := cmd.Execute()
+	re.NoError(err)
+	re.Empty(bufErr)
+
+	var actuals []api.Health
+	err = json.Unmarshal(bufOut.Bytes(), &actuals)
+	re.NoError(err)
+	re.Len(actuals, 1)
+
+	actual := actuals[0]
+	re.Equal(suite.cluster[0].Name(), actual.Name)
+	re.Equal(suite.cluster[0].GetMember().ID(), actual.MemberID)
+	re.True(actual.Health)
+	re.Equal(suite.cluster[0].GetMemberInfo().ClientUrls, actual.ClientUrls)
+}
+
+func (suite *commandTestSuite) TestHotCommand() {
```

Comment:
`pd-ctl` tests already in `tests/pdctl/xxx` package. Could you unify them together?

Reply:
It looks like the tests in `tests/pdctl` are quite comprehensive. This PR feels a bit redundant now, so I am gonna close it.